ffi/capi/src/neo_datetime.rs

```diff
@@ -1012,33 +1606,180 @@ pub mod ffi {
+        DateTimeFormatterLoadError, FixedCalendarDateTimeFormatter,
+    };
+
+    #[cfg(any(feature = "compiled_data", feature = "buffer_provider"))]
+    pub(super) fn map_or_default<Input, Output>(input: Option<Input>) -> Output
+    where
+        Output: From<Input> + Default,
+    {
+        input.map(Output::from).unwrap_or_default()
+    }
+
+    #[cfg(any(feature = "compiled_data", feature = "buffer_provider"))]
+    pub(super) fn formatter_with_zone<Zone>(
+        formatter: &DateTimeFormatter<CompositeDateTimeFieldSet>,
+        locale: &crate::locale_core::ffi::Locale,
+        zone: Zone,
+        load: impl FnOnce(
+            &mut DateTimeNames<icu_datetime::fieldsets::Combo<DateAndTimeFieldSet, Zone>>,
+        ) -> Result<(), icu_datetime::pattern::PatternLoadError>,
+        to_formatter: impl FnOnce(
+            DateTimeNames<Combo<DateAndTimeFieldSet, Zone>>,
+            Combo<DateAndTimeFieldSet, Zone>,
+        ) -> Result<
+            DateTimeFormatter<Combo<DateAndTimeFieldSet, Zone>>,
+            (
+                DateTimeFormatterLoadError,
+                DateTimeNames<Combo<DateAndTimeFieldSet, Zone>>,
+            ),
+        >,
+    ) -> Result<
+        Box<super::ffi::NeoZonedDateTimeFormatter>,
+        crate::errors::ffi::DateTimeFormatterLoadError,
+    >
+    where
+        Zone: DateTimeMarkers + ZoneMarkers,
+        <Zone as DateTimeMarkers>::Z: ZoneMarkers,
+        Combo<DateAndTimeFieldSet, Zone>: DateTimeNamesFrom<CompositeDateTimeFieldSet>,
+        CompositeFieldSet: DateTimeNamesFrom<Combo<DateAndTimeFieldSet, Zone>>,
+    {
+        let prefs = (&locale.0).into();
+        let mut names = DateTimeNames::from_formatter(prefs, formatter.clone())
+            .cast_into_fset::<Combo<DateAndTimeFieldSet, Zone>>();
+        load(&mut names)?;
+        let field_set = formatter
+            .to_field_set_builder()
+            .build_date_and_time()
+            .map_err(|e| {
+                debug_assert!(false, "should be infallible, but got: {e:?}");
+                crate::errors::ffi::DateTimeFormatterLoadError::Unknown
+            })?
+            .zone(zone);
+        let formatter = to_formatter(names, field_set)
+            // This can fail if the locale doesn't match and the fields conflict
+            .map_err(|(e, _)| e)?
+            .cast_into_fset();
+        Ok(Box::new(super::ffi::NeoZonedDateTimeFormatter(formatter)))
+    }
+
+    #[cfg(any(feature = "compiled_data", feature = "buffer_provider"))]
+    pub(super) fn gregorian_formatter_with_zone<Zone>(
+        formatter: &FixedCalendarDateTimeFormatter<Gregorian, CompositeDateTimeFieldSet>,
+        locale: &crate::locale_core::ffi::Locale,
+        zone: Zone,
+        load: impl FnOnce(
+            &mut FixedCalendarDateTimeNames<
+                Gregorian,
+                icu_datetime::fieldsets::Combo<DateAndTimeFieldSet, Zone>,
```

Comment:
issue: Combo should just be imported here IMO

Reply:
Done